extension/src/popup/components/sendPayment/SendTo/index.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect, useCallback } from "react";
+import { useSelector } from "react-redux";
+import debounce from "lodash/debounce";
+import { StrKey } from "stellar-sdk";
+
+import { getAccountBalances } from "@shared/api/internal";
+import { truncatedPublicKey } from "helpers/stellar";
+
+import { navigateTo } from "popup/helpers/navigate";
+import { ROUTES } from "popup/constants/routes";
+import { PopupWrapper } from "popup/basics/PopupWrapper";
+import { settingsNetworkDetailsSelector } from "popup/ducks/settings";
+import { BackButton } from "popup/basics/BackButton";
+import { BottomNav } from "popup/components/BottomNav";
+import { defaultAccountBalances } from "popup/views/Account";
+
+import { Loader } from "@stellar/design-system";
+
+import "../styles.scss";
+
+export const SendTo = ({
+  destination,
+  setDestination,
+}: {
+  destination: string;
+  setDestination: (state: string) => void;
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [isValidAddress, setIsValidAddress] = useState(false);
+  const [destinationBalances, setDestinationBalances] = useState(
+    defaultAccountBalances,
+  );
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+
+  const validatePublicKey = (publicKey: string) => {
+    if (publicKey.startsWith("M")) {
+      // TODO: remove when type is added to stellar-sdk
+      // @ts-ignore
+      if (!StrKey.isValidMed25519PublicKey(publicKey)) {
+        setIsValidAddress(false);
+      }
+    } else if (!StrKey.isValidEd25519PublicKey(publicKey)) {
+      setIsValidAddress(false);
+    } else {
+      setIsValidAddress(true);
+    }
+  };
+
+  const debounceValidate = useCallback(
+    debounce(async (addressInput: string) => {
+      validatePublicKey(addressInput);
+
+      try {
+        const res = await getAccountBalances({
+          publicKey: addressInput,
+          networkDetails,
+        });
+        setDestinationBalances(res);
+      } catch (e) {
+        console.error(e);
+      }
+      setIsLoading(false);
+    }, 2000),
+    [],
+  );
+
+  useEffect(() => {
+    setIsLoading(true);
+    debounceValidate(destination);
+  }, [destination, networkDetails, debounceValidate]);
+
+  return (
+    <PopupWrapper>
+      <BackButton hasBackCopy />
+      <div className="SendTo">
+        <div className="header">Send To</div>
+        <input
+          className="SendTo__input"
+          value={destination}
+          onChange={(e: React.ChangeEvent<any>) =>
+            setDestination(e.target.value)
+          }
+        />
+        {/* TODO - use form validation */}
```

Comment:
I played around with this for a bit but couldn't get it working right. Left these ugly `&&` statements for now but will fix when adding the styling

Reply:
No prob - we can use Formik validators to clean this up a little in the next iteration